frontend/src/components/pages/CampOverview/RemoveCamperModal/index.tsx

```diff
@@ -0,0 +1,158 @@
+          ),
+        );
+        setLoading(false);
+      } else {
+        toast({
+          description: `Unable to retrieve camper information.`,
+          status: "error",
+          duration: 5000,
+          variant: "subtle",
+        });
+      }
+    };
+    getCampersByChargeIdAndSessionId(camper);
+
+    return () => {
+      setRetrievedCampers([]);
+      setCampersToBeDeleted(new Set<Camper>());
+    };
+  }, [camper]);
+
+  const deselectAndClose = () => {
+    setCampersToBeDeleted(new Set<Camper>());
+    setModalStatus(ModalStatus.SELECT);
+    removeModalOnClose();
+  };
+
+  const deletionMessage = () => {
+    if (allCampersToBeDeleted.length >= 2) {
+      let returnString = `${allCampersToBeDeleted[0].firstName} ${allCampersToBeDeleted[0].lastName}, `;
+      for (let i = 1; i < allCampersToBeDeleted.length - 1; i += 1) {
+        returnString += `${allCampersToBeDeleted[i].firstName} ${allCampersToBeDeleted[i].lastName}, `;
+      }
+      returnString += `and ${
+        allCampersToBeDeleted[allCampersToBeDeleted.length - 1].firstName
+      } ${
+        allCampersToBeDeleted[allCampersToBeDeleted.length - 1].lastName
+      } have been removed.`;
+      return returnString;
+    }
+    return `${allCampersToBeDeleted[0].firstName} ${allCampersToBeDeleted[0].lastName} has been removed.`;
+  };
+
+  // Delete the selected campers.
+  const submitDeletion = async () => {
+    const deletionResponse = await CamperAPIClient.deleteMultipleCampersById(
+      allCampersToBeDeleted.map((currentCamper: Camper) => currentCamper.id),
+    );
+
+    if (deletionResponse) {
+      toast({
+        description: deletionMessage(),
+        status: "success",
+        duration: 5000,
+        variant: "subtle",
+      });
+    } else {
+      toast({
+        description: `Unable to remove the selected campers.`,
+        status: "error",
+        duration: 5000,
+        variant: "subtle",
+      });
+    }
+    handleRefetch();
+  };
+
+  return (
+    <Modal
+      isOpen={removeModalIsOpen}
+      onClose={deselectAndClose}
+      isCentered
+      preserveScrollBarGap
+      scrollBehavior="inside"
+    >
+      <ModalOverlay />
+
+      {modalStatus === ModalStatus.SELECT &&
+        retrievedCampers.length > 0 &&
+        !loading && (
+          <SelectModal
+            camper={camper}
+            campersToBeDeleted={campersToBeDeleted}
+            setCampersToBeDeleted={setCampersToBeDeleted}
+            retrievedCampers={retrievedCampers}
+            deselectAndClose={deselectAndClose}
+            setStatusAsConfirm={() => setModalStatus(ModalStatus.CONFIRM)}
+          />
+        )}
+
+      {(modalStatus === ModalStatus.CONFIRM || retrievedCampers.length === 0) &&
+        !loading && (
+          <ConfirmModal
+            allCampersToBeDeleted={allCampersToBeDeleted}
+            deselectAndClose={deselectAndClose}
+            submitDeletion={submitDeletion}
+          />
+        )}
```

Comment:
I extracted the two boolean expressions into their own functions, I think that increases readability (pls review), but I still want to keep both modal elements themselves in the actual return statement. Also I think a dedicated loading case is unneccesary since we aren't doing anything while `loading` is true, we only want to display things when `loading` is false.